app/queries/arel_helpers.rb

```diff
@@ -0,0 +1,34 @@
+module Queries
+  module ArelHelpers
+    CTE = Struct.new(:table, :composed_cte)
+
+    def table(table)
+      Arel::Table.new(table, ActiveRecord::Base)
+    end
+
+    def cte(scope, as:)
```

Comment:
Given that Arel isn't that well-known, most people (including me) won't know what a CTE is. It would be helpful to have a comment explaining what this is for and how to use it.